src/main/java/simplifier/security/TokenAuthenticationFilter.java

```diff
@@ -0,0 +1,47 @@
+package simplifier.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+public class TokenAuthenticationFilter extends GenericFilterBean {
+
+    private final String AUTH_HEADER = "Authorization";
+
+    private TokenHelper tokenHelper;
+
+    @Autowired
+    public void setTokenHelper(TokenHelper tokenHelper) {
+        this.tokenHelper = tokenHelper;
+    }
+
+    private String getToken(HttpServletRequest request) {
```

Comment:
private method should be below public ones. Use [this](http://www.oracle.com/technetwork/java/codeconventions-150003.pdf) as a reference